arviz/data/inference_data.py

```diff
@@ -417,6 +448,39 @@ def stack(
             A new InferenceData object by default.
             When `inplace==True` perform selection in-place and return `None`
 
+        Examples
+        --------
+        Use ``stack`` to stack any number of existing dimensions into a single new dimension.. We first check the
+        original object:
+
+        .. ipython::
+
+            In [1]: import arviz as az
+               ...: datadict = {
+                    "a": np.random.randn(100),
+                    "b": np.random.randn(1, 100, 10),
+                    "c": np.random.randn(1, 100, 3, 4),
+                    }
+               ...: coords = {
+                        "c1": np.arange(3),
+                        "c99": np.arange(4),
+                        "b1": np.arange(10),
+                    }
+               ...: dims = {"c": ["c1", "c99"], "b": ["b1"]}
+               ...: dataset = from_dict(posterior=datadict, posterior_predictive=datadict, coords=coords, dims=dims)
```

Comment:
you are importing arviz as az, thus this should be `az.from_dict`, I also recommend not using dataset due to confusion with xarray datasets, better use `data`, `idata`, the name of the model like radon or eight_schools...